elastic/beats
# Add daemonset name in pods controlled by DaemonSets

## What does this PR do?

This PR adds `kubernetes.daemonset.name` in Pods' metadata.

## Why is it important?

It enriches the documents of pods belonging to `DaemonSets` with the DaemonSet name.

## Checklist





## How to test this PR locally

1.  Deploy a sample DaemonSet on k8s: `kubectl apply -f https://k8s.io/examples/controllers/daemonset.yaml`
2. Deploy Metricbeat ([instructions](https://github.com/elastic/beats/tree/master/metricbeat/module/kubernetes/_meta/test#readme)) on k8s and verify that events shipped by pod and state_pod metricset include `kubernetes.daemonset.name `. As an example see screenshots below.

## Related issues

- Closes https://github.com/elastic/beats/issues/25816 


## Screenshots

<img width="1792" alt="Screenshot 2021-07-09 at 1 31 55 PM" src="https://user-images.githubusercontent.com/26270880/125065220-3c970080-e0ba-11eb-8484-55cc6ee58342.png">
